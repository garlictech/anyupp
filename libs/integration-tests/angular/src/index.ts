export * from './lib/integration-tests-angular.module';
