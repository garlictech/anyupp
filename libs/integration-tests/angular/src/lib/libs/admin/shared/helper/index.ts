import { Auth } from 'aws-amplify';

import { awsConfig } from '@bgap/crud-gql/api';
import {
  getCognitoUsername,
  testAdminUserPassword,
} from '@bgap/shared/fixtures';

export const signInToCognito = async () => {
  Auth.configure({
    ...awsConfig,
    authenticationFlowType: 'USER_PASSWORD_AUTH',
    aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  });

  await Auth.signIn(getCognitoUsername('monad'), testAdminUserPassword);
};

export const signOutFromCognito = async () => {
  await Auth.signOut();
};
