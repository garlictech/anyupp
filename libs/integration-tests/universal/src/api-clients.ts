import {
  awsConfig,
  getCrudSdkForIAM,
  getCrudSdkForUserPool,
} from '@bgap/crud-gql/api';
import {
  getAnyuppSdkForIAM,
  getAnyuppSdkForUserPool,
} from '@bgap/anyupp-gql/api';
import { Auth } from 'aws-amplify';
import { from } from 'rxjs';
import { map } from 'rxjs/operators';

Auth.configure({
  ...awsConfig,
  // See: https://github.com/aws-amplify/amplify-js/issues/6552#issuecomment-682259256
  authenticationFlowType: 'USER_PASSWORD_AUTH',
  aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
});

export const createAuthenticatedCrudSdk = (
  userName: string,
  password: string,
) => from(Auth.signIn(userName, password)).pipe(map(getCrudSdkForUserPool));

export const createAuthenticatedAnyuppSdk = (
  userName: string,
  password: string,
) => from(Auth.signIn(userName, password)).pipe(map(getAnyuppSdkForUserPool));

export const createIamCrudSdk = () =>
  getCrudSdkForIAM(
    process.env.AWS_ACCESS_KEY_ID || '',
    process.env.AWS_SECRET_ACCESS_KEY || '',
  );

export const createIamAnyuppSdk = () =>
  getAnyuppSdkForIAM(
    process.env.AWS_ACCESS_KEY_ID || '',
    process.env.AWS_SECRET_ACCESS_KEY || '',
  );
