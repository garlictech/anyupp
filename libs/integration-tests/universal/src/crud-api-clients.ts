import {
  awsConfig,
  getCrudSdkForIAM,
  getCrudSdkForUserPool,
} from '@bgap/crud-gql/api';
import { Auth } from 'aws-amplify';
import { from, Observable } from 'rxjs';
import { map, mapTo } from 'rxjs/operators';

export const createAuthenticatedCrudSdk = (
  userName: string,
  password: string,
) => {
  Auth.configure({
    ...awsConfig,
    // See: https://github.com/aws-amplify/amplify-js/issues/6552#issuecomment-682259256
    authenticationFlowType: 'USER_PASSWORD_AUTH',
    aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  });

  return from(Auth.signIn(userName, password)).pipe(map(getCrudSdkForUserPool));
};

export const createIamSdk = () =>
  getCrudSdkForIAM(
    process.env.AWS_ACCESS_KEY_ID || '',
    process.env.AWS_SECRET_ACCESS_KEY || '',
  );
