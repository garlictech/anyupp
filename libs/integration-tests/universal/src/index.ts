export { testAdminUsername, testAdminUserPassword } from './lib/common';
