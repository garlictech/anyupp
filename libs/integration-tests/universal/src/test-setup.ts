(global as any).WebSocket = require('ws');
require('es6-promise').polyfill();
require('isomorphic-fetch');
