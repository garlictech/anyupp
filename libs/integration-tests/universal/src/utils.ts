import * as R from 'ramda';
import { pipe } from 'fp-ts/lib/function';

export const ES_DELAY = 3000; //ms
export const DYNAMO_DELAY = 1000; //ms

// Replace UUID-s with a static string for jest matchers
export const maskV4UuidIds = <T>(data: T) =>
  pipe(
    data,
    JSON.stringify,
    R.replace(
      /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/g,
      'MASKED ID',
    ),
    (d: string) => JSON.parse(d) as T,
  );
