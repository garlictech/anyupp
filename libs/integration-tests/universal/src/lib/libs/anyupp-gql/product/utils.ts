import * as CrudApi from '@bgap/crud-gql/api';
import { throwError } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { throwIfEmptyValue } from '@bgap/shared/utils';

export const getChainProduct = (sdk: CrudApi.CrudSdk, productId: string) => {
  return sdk.GetChainProduct({ id: productId }).pipe(
    throwIfEmptyValue<CrudApi.ChainProduct>(),
    catchError(err => {
      return throwError('ChainProdcuct is missing');
    }),
  );
};

export const getGroupProduct = (sdk: CrudApi.CrudSdk, productId: string) => {
  return sdk.GetGroupProduct({ id: productId }).pipe(
    throwIfEmptyValue<CrudApi.GroupProduct>(),
    catchError(err => {
      return throwError('GroupProdcuct is missing');
    }),
  );
};

export const getUnitProduct = (sdk: CrudApi.CrudSdk, productId: string) => {
  return sdk.GetUnitProduct({ id: productId }).pipe(
    throwIfEmptyValue<CrudApi.UnitProduct>(),
    catchError(err => {
      return throwError('UnitProdcuct is missing');
    }),
  );
};
