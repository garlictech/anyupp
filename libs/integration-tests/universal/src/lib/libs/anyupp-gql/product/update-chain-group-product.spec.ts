import Auth from '@aws-amplify/auth';
import * as AnyuppApi from '@bgap/anyupp-gql/api';
import * as CrudApi from '@bgap/crud-gql/api';
import {
  productFixture,
  testAdminUsername,
  testAdminUserPassword,
  testIdPrefix,
} from '@bgap/shared/fixtures';
import { RequiredId } from '@bgap/shared/types';
import { productRequestHandler } from 'libs/anyupp-gql/backend/src';
import { combineLatest, concat, defer, iif, of, throwError } from 'rxjs';
import {
  catchError,
  delay,
  switchMap,
  takeLast,
  toArray,
} from 'rxjs/operators';
import {
  createAuthenticatedAnyuppSdk,
  createIamCrudSdk,
} from '../../../../api-clients';
import {
  createTestChainProduct,
  deleteTestChainProduct,
} from '../../../seeds/chain-product';
import {
  createTestGroupProduct,
  deleteTestGroupProduct,
} from '../../../seeds/group-product';
import {
  createTestUnitProduct,
  deleteTestUnitProduct,
} from '../../../seeds/unit-product';
import { getChainProduct, getGroupProduct } from './utils';

const TEST_NAME = 'UPDATE_CHAIN_GROUP_PROD_';
const DEBUG_MODE_TEST_WITH_LOCALE_CODE = false;
const DYNAMODB_OPERATION_DELAY = 3000;
const unitId_01 = `${testIdPrefix}${TEST_NAME}UNIT_ID_01`;
const unitId_02 = `${testIdPrefix}${TEST_NAME}UNIT_ID_02`;

// PRODUCTS to create
const chainProduct_01: RequiredId<CrudApi.CreateChainProductInput> = {
  ...productFixture.chainProductInputBase,
  id: `${testIdPrefix}${TEST_NAME}chainProduct_01`,
};
const chainProduct_02_WITH_NO_CHILD: RequiredId<CrudApi.CreateChainProductInput> =
  {
    ...productFixture.chainProductInputBase,
    id: `${testIdPrefix}${TEST_NAME}chainProduct_02`,
  };
const groupProduct_01: RequiredId<CrudApi.CreateGroupProductInput> = {
  ...productFixture.groupProductInputBase,
  id: `${testIdPrefix}${TEST_NAME}groupProduct_01`,
  parentId: chainProduct_01.id,
};
const groupProduct_02: RequiredId<CrudApi.CreateGroupProductInput> = {
  ...productFixture.groupProductInputBase,
  id: `${testIdPrefix}${TEST_NAME}groupProduct_02`,
  parentId: chainProduct_01.id,
};
const groupProduct_03_WITH_NO_CHILD: RequiredId<CrudApi.CreateGroupProductInput> =
  {
    ...productFixture.groupProductInputBase,
    id: `${testIdPrefix}${TEST_NAME}groupProduct_03`,
    parentId: chainProduct_01.id,
  };

const unitProduct_u01_gp01_01: RequiredId<CrudApi.CreateUnitProductInput> = {
  ...productFixture.unitProductInputBase,
  id: `${testIdPrefix}${TEST_NAME}unitProduct_01`,
  parentId: groupProduct_01.id,
  unitId: unitId_01,
};
const unitProduct_u01_gp02_02: RequiredId<CrudApi.CreateUnitProductInput> = {
  ...productFixture.unitProductInputBase,
  id: `${testIdPrefix}${TEST_NAME}unitProduct_02`,
  parentId: groupProduct_02.id,
  unitId: unitId_01,
};
const unitProduct_u02_gp01_03: RequiredId<CrudApi.CreateUnitProductInput> = {
  ...productFixture.unitProductInputBase,
  id: `${testIdPrefix}${TEST_NAME}unitProduct_03`,
  parentId: groupProduct_01.id,
  unitId: unitId_02,
};

describe('Update Chain and Group product tests', () => {
  let authAnyuppSdk: AnyuppApi.AnyuppSdk;

  const iamCrudSdk = createIamCrudSdk();

  const cleanup = concat(
    // CleanUP
    deleteTestUnitProduct(unitProduct_u01_gp01_01.id, iamCrudSdk),
    deleteTestUnitProduct(unitProduct_u01_gp02_02.id, iamCrudSdk),
    deleteTestUnitProduct(unitProduct_u02_gp01_03.id, iamCrudSdk),
    deleteTestGroupProduct(groupProduct_01.id, iamCrudSdk),
    deleteTestGroupProduct(groupProduct_02.id, iamCrudSdk),
    deleteTestGroupProduct(groupProduct_03_WITH_NO_CHILD.id, iamCrudSdk),
    deleteTestChainProduct(chainProduct_01.id, iamCrudSdk),
    deleteTestChainProduct(chainProduct_02_WITH_NO_CHILD.id, iamCrudSdk),
  ).pipe(toArray());

  beforeAll(async () => {
    authAnyuppSdk = await createAuthenticatedAnyuppSdk(
      testAdminUsername,
      testAdminUserPassword,
    )
      .toPromise()
      .then(x => x.authAnyuppSdk);

    await cleanup
      .pipe(
        delay(DYNAMODB_OPERATION_DELAY),
        takeLast(1),
        switchMap(() =>
          // Seeding
          combineLatest([
            createTestChainProduct(chainProduct_01, iamCrudSdk),
            createTestChainProduct(chainProduct_02_WITH_NO_CHILD, iamCrudSdk),
            createTestGroupProduct(groupProduct_01, iamCrudSdk),
            createTestGroupProduct(groupProduct_02, iamCrudSdk),
            createTestGroupProduct(groupProduct_03_WITH_NO_CHILD, iamCrudSdk),
            createTestUnitProduct(unitProduct_u01_gp01_01, iamCrudSdk),
            createTestUnitProduct(unitProduct_u01_gp02_02, iamCrudSdk),
            createTestUnitProduct(unitProduct_u02_gp01_03, iamCrudSdk),
          ]),
        ),
        takeLast(1),
        delay(DYNAMODB_OPERATION_DELAY),
        catchError(err => {
          console.error('BEFORE HOOK ERROR');
          return throwError(err);
        }),
      )
      .toPromise();
  }, 25000);

  afterAll(async () => {
    await cleanup.toPromise();
    await Auth.signOut();
  });

  describe('GroupProduct Update', () => {
    it('should throw in case of an invalid input', done => {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const input: any = { foo: 'NOT A VALID INPUT' };
      iif(
        () => DEBUG_MODE_TEST_WITH_LOCALE_CODE,
        defer(() =>
          productRequestHandler({
            crudSdk: iamCrudSdk,
            regenerateUnitDataHandler: jest.fn(),
          }).updateGroupProduct({ input }),
        ),
        authAnyuppSdk.UpdateGroupProduct({ input }),
      ).subscribe({
        next() {
          console.error(`${TEST_NAME}Test ERROR`, 'SHOULD NOT SUCCEED');
        },
        error(err) {
          expect(err).toMatchSnapshot();
          done();
        },
      });
    }, 15000);

    it('should update the groupProduct in the database and call the regenereation logic for every unit', done => {
      const regenMockHandler = jest.fn().mockReturnValue(of(true));
      const input: AnyuppApi.UpdateGroupProductInput = {
        id: groupProduct_01.id,
        tax: 999,
      };

      defer(() =>
        productRequestHandler({
          crudSdk: iamCrudSdk,
          regenerateUnitDataHandler: regenMockHandler,
        }).updateGroupProduct({
          input,
        }),
      )
        .pipe(
          delay(DYNAMODB_OPERATION_DELAY),
          switchMap(product => getGroupProduct(iamCrudSdk, product.id)),
        )
        .subscribe({
          next(result) {
            const { createdAt, updatedAt, ...product } = result;
            expect(createdAt).not.toBeUndefined();
            expect(updatedAt).not.toBeUndefined();
            expect(product.id).toEqual(groupProduct_01.id);
            expect(product.tax).toEqual(999);
            expect(product).toMatchSnapshot('TAX SHOULD BE 999');

            // Regeneration check
            expect(regenMockHandler).toBeCalledTimes(2);
            const allTheCallArguments = regenMockHandler.mock.calls.map(
              call => call[0],
            );
            expect(allTheCallArguments).toHaveLength(2);
            expect(allTheCallArguments).toContain(
              unitProduct_u01_gp01_01.unitId,
            );
            expect(allTheCallArguments).toContain(
              unitProduct_u02_gp01_03.unitId,
            );
            done();
          },
        });
    }, 25000);

    it('should NOT fail in case there is no extended UnitProduct for the given GroupProduct', done => {
      const regenMockHandler = jest.fn().mockReturnValue(of(true));
      const input: AnyuppApi.UpdateGroupProductInput = {
        id: groupProduct_03_WITH_NO_CHILD.id,
      };

      iif(
        () => DEBUG_MODE_TEST_WITH_LOCALE_CODE,
        defer(() =>
          productRequestHandler({
            crudSdk: iamCrudSdk,
            regenerateUnitDataHandler: regenMockHandler,
          }).updateGroupProduct({
            input,
          }),
        ),
        authAnyuppSdk.UpdateGroupProduct({ input }),
      ).subscribe({
        next(product) {
          expect(product).not.toBeUndefined();
          expect(product.id).toEqual(groupProduct_03_WITH_NO_CHILD.id);

          expect(regenMockHandler).not.toBeCalled();
          done();
        },
        error(err) {
          console.error(`${TEST_NAME}Test ERROR`, err);
        },
      });
    }, 25000);
  });

  describe('ChainProduct Update', () => {
    it('should throw in case of an invalid input', done => {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const input: any = { foo: 'NOT A VALID INPUT' };
      iif(
        () => DEBUG_MODE_TEST_WITH_LOCALE_CODE,
        defer(() =>
          productRequestHandler({
            crudSdk: iamCrudSdk,
            regenerateUnitDataHandler: jest.fn(),
          }).updateChainProduct({ input }),
        ),
        authAnyuppSdk.UpdateChainProduct({ input }),
      ).subscribe({
        next() {
          console.error(`${TEST_NAME}Test ERROR`, 'SHOULD NOT SUCCEED');
        },
        error(err) {
          expect(err).toMatchSnapshot();
          done();
        },
      });
    }, 15000);

    it('should update the chainProduct in the database and call the regenereation logic for every unit', done => {
      const regenMockHandler = jest.fn().mockReturnValue(of(true));
      const input: AnyuppApi.UpdateChainProductInput = {
        id: chainProduct_01.id,
        description: { en: 'UPDATED_DESCRIPTION' },
      };

      defer(() =>
        productRequestHandler({
          crudSdk: iamCrudSdk,
          regenerateUnitDataHandler: regenMockHandler,
        }).updateChainProduct({
          input,
        }),
      )
        .pipe(
          delay(DYNAMODB_OPERATION_DELAY),
          switchMap(() => getChainProduct(iamCrudSdk, input.id)),
        )
        .subscribe({
          next(result) {
            const { createdAt, updatedAt, ...product } = result;
            expect(createdAt).not.toBeUndefined();
            expect(updatedAt).not.toBeUndefined();
            expect(product.id).toEqual(chainProduct_01.id);
            expect(product.description?.en).toEqual(input.description?.en);
            expect(product).toMatchSnapshot(
              'description SHOULD BE UPDATED_DESCRIPTION',
            );

            // Regeneration check
            expect(regenMockHandler).toBeCalledTimes(3);
            const allTheCallArguments = regenMockHandler.mock.calls.map(
              call => call[0],
            );
            expect(allTheCallArguments).toHaveLength(3);
            expect(allTheCallArguments).toContain(
              unitProduct_u01_gp01_01.unitId,
            );
            expect(allTheCallArguments).toContain(
              unitProduct_u01_gp02_02.unitId,
            );
            expect(allTheCallArguments).toContain(
              unitProduct_u02_gp01_03.unitId,
            );
            done();
          },
        });
    }, 25000);

    it('should NOT fail in case there is no extended UnitProduct for the given ChainProduct', done => {
      const regenMockHandler = jest.fn().mockReturnValue(of(true));
      const input: AnyuppApi.UpdateChainProductInput = {
        id: chainProduct_02_WITH_NO_CHILD.id,
      };

      iif(
        () => DEBUG_MODE_TEST_WITH_LOCALE_CODE,
        defer(() =>
          productRequestHandler({
            crudSdk: iamCrudSdk,
            regenerateUnitDataHandler: regenMockHandler,
          }).updateChainProduct({
            input,
          }),
        ),
        authAnyuppSdk.UpdateChainProduct({ input }),
      ).subscribe({
        next(product) {
          expect(product).not.toBeUndefined();
          expect(product.id).toEqual(chainProduct_02_WITH_NO_CHILD.id);

          expect(regenMockHandler).not.toBeCalled();
          done();
        },
        error(err) {
          console.error(`${TEST_NAME}Test ERROR`, err);
        },
      });
    }, 25000);
  });
});
