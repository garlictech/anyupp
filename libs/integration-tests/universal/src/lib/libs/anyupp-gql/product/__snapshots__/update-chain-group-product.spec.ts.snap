// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Chain and Group product tests ChainProduct Update should throw in case of an invalid input 1`] = `[Error: GraphQL error: The variables input contains a field name 'foo' that is not defined for input object type 'UpdateChainProductInput' ]`;

exports[`Update Chain and Group product tests ChainProduct Update should update the chainProduct in the database and call the regenereation logic for every unit: description SHOULD BE UPDATED_DESCRIPTION 1`] = `
Object {
  "allergens": Array [
    "egg",
    "gluten",
  ],
  "chainId": "chainId_",
  "configSets": null,
  "description": Object {
    "de": null,
    "en": "UPDATED_DESCRIPTION",
    "hu": null,
  },
  "id": "test_UPDATE_CHAIN_GROUP_PROD_chainProduct_01",
  "image": "IMAGE",
  "isVisible": true,
  "name": Object {
    "de": null,
    "en": "CHAIN_PRODUCT",
    "hu": null,
  },
  "productCategoryId": "productCategoryId_",
  "productType": "PRODUCT_TYPE",
  "variants": Array [
    Object {
      "availabilities": Array [
        Object {
          "dayFrom": "",
          "dayTo": "",
          "price": 0.5,
          "timeFrom": "",
          "timeTo": "",
          "type": "A",
        },
      ],
      "id": "test_ChainProductVariant_id_1",
      "isAvailable": true,
      "pack": Object {
        "size": 1,
        "unit": "CHAIN_UNIT",
      },
      "position": 1,
      "price": 10,
      "refGroupPrice": 10,
      "variantName": Object {
        "de": null,
        "en": "VARIANT_NAME_1",
        "hu": null,
      },
    },
    Object {
      "availabilities": Array [
        Object {
          "dayFrom": "",
          "dayTo": "",
          "price": 1,
          "timeFrom": "",
          "timeTo": "",
          "type": "A",
        },
      ],
      "id": "test_ChainProductVariant_id_2",
      "isAvailable": true,
      "pack": Object {
        "size": 2,
        "unit": "CHAIN_UNIT",
      },
      "position": 2,
      "price": 20,
      "refGroupPrice": 20,
      "variantName": Object {
        "de": null,
        "en": "VARIANT_NAME_2",
        "hu": null,
      },
    },
  ],
}
`;

exports[`Update Chain and Group product tests GroupProduct Update should throw in case of an invalid input 1`] = `[Error: GraphQL error: The variables input contains a field name 'foo' that is not defined for input object type 'UpdateGroupProductInput' ]`;

exports[`Update Chain and Group product tests GroupProduct Update should update the groupProduct in the database and call the regenereation logic for every unit: TAX SHOULD BE 999 1`] = `
Object {
  "chainId": "chainId_",
  "configSets": null,
  "groupId": "groupId_",
  "id": "test_UPDATE_CHAIN_GROUP_PROD_groupProduct_01",
  "isVisible": true,
  "parentId": "test_UPDATE_CHAIN_GROUP_PROD_chainProduct_01",
  "tax": 999,
  "variants": Array [
    Object {
      "availabilities": Array [
        Object {
          "dayFrom": "",
          "dayTo": "",
          "price": 1,
          "timeFrom": "",
          "timeTo": "",
          "type": "A",
        },
      ],
      "id": "test_GroupProductVariant_id_1",
      "isAvailable": true,
      "pack": Object {
        "size": 1,
        "unit": "GROUP_UNIT",
      },
      "position": 1,
      "price": 20,
      "refGroupPrice": 20,
      "variantName": Object {
        "de": null,
        "en": "VARIANT_NAME_1",
        "hu": null,
      },
    },
    Object {
      "availabilities": Array [
        Object {
          "dayFrom": "",
          "dayTo": "",
          "price": 2,
          "timeFrom": "",
          "timeTo": "",
          "type": "A",
        },
      ],
      "id": "test_GroupProductVariant_id_2",
      "isAvailable": true,
      "pack": Object {
        "size": 2,
        "unit": "GROUP_UNIT",
      },
      "position": 2,
      "price": 40,
      "refGroupPrice": 40,
      "variantName": Object {
        "de": null,
        "en": "VARIANT_NAME_2",
        "hu": null,
      },
    },
  ],
}
`;
